src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -263,9 +263,11 @@ function assertValidProps(component, props) {
   }
   invariant(
     props.style == null || typeof props.style === 'object',
-    'The `style` prop expects a mapping from style properties to values, ' +
+    'A dom node has incorrect inline style syntax.' +
+     ' The `style` prop expects a mapping from style properties to values, ' +
     'not a string. For example, style={{marginRight: spacing + \'em\'}} when ' +
-    'using JSX.'
+    'using JSX. %s',
```

Comment:
The addendum will start with a space, you should not be adding one yourself here.